stdlib/public/runtime/Metadata.cpp

```diff
@@ -3040,18 +3041,38 @@ void swift::swift_initRawStructMetadata(StructMetadata *structType,
   auto alignMask = likeTypeLayout->flags.getAlignmentMask();
   auto extraInhabitantCount = likeTypeLayout->extraInhabitantCount;
 
-  // If our count is greater than or equal 0, we're dealing an array like layout.
-  if (count >= 0) {
-    stride *= count;
+  if (isRawLayoutArray(rawLayoutFlags)) {
+    // Our count value may be negative, so use 0 if that's the case.
+    stride *= std::max(count, (intptr_t)0);
```

Comment:
Existing binaries are not going to pass a value for `rawLayoutFlags` and this code will receive a garbage value. Changing `count` to an `intptr_t` is also fishy since existing code may not extend the parameter value to occupy the full word size on 64-bit platforms. The safest thing to do would be to make this a new `swift_initRawStructMetadata2` entry point. If we're absolutely sure the calling convention for the existing arguments isn't perturbed on any ABI-stable platform by the parameter changes, there would at least need to be a flag in `structLayoutFlags` to indicate that the new parameters are populated.

Reply:
Ok, I went ahead and added a new `swift_initRawStructMetadata2` entry just to be super safe.